ledger/participant-integration-api/src/main/scala/platform/apiserver/services/admin/ApiParticipantPruningService.scala

```diff
@@ -39,65 +44,86 @@ final class ApiParticipantPruningService private (
 
   override def bindService(): ServerServiceDefinition =
     ParticipantPruningServiceGrpc.bindService(this, executionContext)
 
   override def close(): Unit = ()
 
   override def prune(request: PruneRequest): Future[PruneResponse] =
-    withEnrichedLoggingContext(traceId(telemetry.traceIdFromGrpcContext)) {
-      implicit loggingContext =>
-        val submissionIdOrErr = Ref.SubmissionId
-          .fromString(
-            if (request.submissionId.nonEmpty) request.submissionId else UUID.randomUUID().toString
-          )
-          .left
-          .map(err =>
-            invalidArgument(s"submission_id $err")(
-              contextualizedErrorLogger(request.submissionId)
-            )
+    withValidatedSubmissionId(request) { submissionId =>
+      LoggingContext.withEnrichedLoggingContext(
+        logging.submissionId(submissionId),
+        traceId(telemetry.traceIdFromGrpcContext),
+      ) { implicit loggingContext =>
+        implicit val contextualizedErrorLogger: ContextualizedErrorLogger =
+          new DamlContextualizedErrorLogger(
+            logger = logger,
+            loggingContext = loggingContext,
+            correlationId = Some(submissionId),
           )
 
-        submissionIdOrErr.fold(
-          t => Future.failed(ValidationLogger.logFailure(request, t)),
-          submissionId =>
-            LoggingContext.withEnrichedLoggingContext(logging.submissionId(submissionId)) {
-              implicit loggingContext =>
-                logger.info(s"Pruning up to ${request.pruneUpTo}")
-                (for {
-
-                  pruneUpTo <- validateRequest(request)(
-                    loggingContext,
-                    contextualizedErrorLogger(submissionId),
-                  )
-
-                  // If write service pruning succeeds but ledger api server index pruning fails, the user can bring the
-                  // systems back in sync by reissuing the prune request at the currently specified or later offset.
-                  _ <- Tracked.future(
-                    metrics.daml.services.pruning.pruneCommandStarted,
-                    metrics.daml.services.pruning.pruneCommandCompleted,
-                    pruneWriteService(pruneUpTo, submissionId, request.pruneAllDivulgedContracts),
-                  )(MetricsContext(("phase", "underlyingLedger")))
-
-                  pruneResponse <- Tracked.future(
-                    metrics.daml.services.pruning.pruneCommandStarted,
-                    metrics.daml.services.pruning.pruneCommandCompleted,
-                    pruneLedgerApiServerIndex(
-                      pruneUpTo,
-                      request.pruneAllDivulgedContracts,
-                    ),
-                  )(MetricsContext(("phase", "ledgerApiServerIndex")))
-
-                } yield pruneResponse).andThen(logger.logErrorsOnCall[PruneResponse])
-            },
-        )
+        ifNotRunning {
```

Comment:
Is there currently no protection against multiple requests running in parallel ?

Reply:
No and am not sure why not. Was it a relaxation just because the lower-level semantics allowed it or because there was an actual use case in focus at some point? @mziolekda maybe knows